### PR TITLE
Fix #10609

### LINKED ISSUE #10609: Segfault when refining mask on gaussian blur retouch
**To Reproduce**
1. Open the Retouch module
2. Select "Gaussian Blur"
3. Pick a shape (verified with circle and ellipse)
3. Place the shape partially off the image
4. Click on the "Parametric mask" button
5. Drag the "details threshold" slider

At this point, darktable segfaults.

**Expected behavior**
Gaussian blur is applied only to parts of the image as constrained by the detail threshold

**Platform**

* darktable version : 3.6.1-1
* OS                : Linux, kernel 5.15.0-2-amd64 
* Linux - Distro    : Debian testing
* Memory            : 16 GB 
* Graphics card     : Intel Corporation UHD Graphics 620 (rev 07)
* Graphics driver   : i915
* OpenCL installed  : no
* OpenCL activated  : no
* Xorg              : 1:7.7+23
* Desktop           : xmonad
* GTK+              : 4.4.1+ds1-3
* gcc               : 11.2.0-12

compile options:
```
  bit depth is 64 bit
  normal build
  SSE2 optimized codepath enabled
  OpenMP support enabled
  OpenCL support enabled
  Lua support enabled, API version 7.0.0
  Colord support enabled
  gPhoto2 support enabled
  GraphicsMagick support enabled
  ImageMagick support disabled
  OpenEXR support enabled
```

**Additional context**
 - Can you reproduce with a RAW or Jpeg or both? **darktable doesn't show the 'detail threshold' slider on jpegs**
 - Are the steps above reproducible with a fresh edit (i.e. after discarding history)? **yes**
 - Is the issue still present using an empty/new config-dir (e.g. start darktable with --configdir "/tmp")? **no**
 - Do you use lua scripts? **no**

Backtrace:
[darktable_bt_SSWIE1.txt](https://github.com/darktable-org/darktable/files/7699958/darktable_bt_SSWIE1.txt)

darktablerc: (personal info redacted with `XXX`)
[darktablerc.txt](https://github.com/darktable-org/darktable/files/7699988/darktablerc.txt)

### PR DESCRIPTION
**Pull request by jenshannoschwalm**
As reported there is an issue with details mask in certain situations as described in issue #10609.

I could find two bugs:
1. In the refining blending loop a wrong size was used (had used roi_in instead of roi_out)
2. a possible double free situation, i had never seen this (it should not be possible in the code flow) but better fixed now.

While tracking this down i was simply not sure where to find the problem so i introduced a number of debugging info to be activated via -d masks option.

Couldn't trigger the crash any more and for me the fix is no workaround but probably just right. 

@TurboGit should leav you some time to test before release. Sorry i stepped in that late ...

@MStraeten could you have a look too? I am not sure about your crop problem though

Probably fixes #10609

### CODE REVIEW

File: src/develop/pixelpipe_hb.c [hunk 1]
```diff
@@ -2589,15 +2618,28 @@ float *dt_dev_distort_detail_mask(const dt_dev_pixelpipe_t *pipe, float *src, co
...
           resmask = tmp;
           if(inmask != src) dt_free_align(inmask);
           inmask = tmp;
         }
+        else if(!module->module->distort_mask &&
+                (module->processed_roi_in.width != module->processed_roi_out.width ||
+                 module->processed_roi_in.height != module->processed_roi_out.height ||
+                 module->processed_roi_in.x != module->processed_roi_out.x ||
+                 module->processed_roi_in.y != module->processed_roi_out.y))
+              printf("FIXME: module `%s' changed the roi from %d x %d @ %d / %d to %d x %d | %d / %d but doesn't have "
```
Comment (TurboGit) [inline src/develop/pixelpipe_hb.c:2632]: Should probably fprintf(stderr, ...

Comment (jenshannoschwalm) [reply src/develop/pixelpipe_hb.c:2632]: right

**Comment by MStraeten** [comment]
fixes issue with crop module through retouch shape
quick test with heal, crop, blur + circle, path

**Comment by jenshannoschwalm** [comment]
Thanks for the quick test

**Comment by MStraeten** [comment]
unfortunately doesn't fix the scenario from #10422 - so thats a quite different issue (at least details mask is not used) but also harder to get reproducible scenarios.

**Comment by jenshannoschwalm** [comment]
I also think that is a different thing. I couldn't reproduce so far btw.

**Review by TurboGit — APPROVED** [review]
On my side I cannot reproduce the issue with or without OpenCL.
So I'll do a blind merge as OP have reported this to fix the issue.
Thanks a lot for the quick fix @jenshannoschwalm!

**Comment by jenshannoschwalm** [comment]
Please note for reproducing the issue you have to resize / rescale the darkroom window.

@TurboGit shall we later remove the debug Infos?

**Comment by TurboGit** [comment]
> shall we later remove the debug Infos?

Not sure, this is controlled by "-d masks"... So looks ok to me.

**Pull request merged**